### PR TITLE
Update SAMP hub web profile for Private Network Access

### PR DESCRIPTION
**Pull request by mbtaylor**
The CORS HTTP server handling localhost web profile requests now does header manipulation as required by the Private Network Access proposal (https://wicg.github.io/private-network-access/). At present, the effect is to prevent recent versions of Google Chrome (e.g. 122) from complaining "Ensure private network requests are only made to resources that allow them"; future versions of Chrome or possibly other browsers may block such requests.

### Description

This pull request is to address ...

Fixes #<Issue Number>

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: docs/changes/samp/16193.feature.rst
Comment (pllim) [inline docs/changes/samp/16193.feature.rst:1]: @saimn , what is our policy to put URL in change log again? We don't want to have to go back and fix history when it breaks in the future, right?

**Review by pllim — COMMENTED** [review]
Thanks!

Does this need user doc update too, in addition to change log?

**Comment by mbtaylor** [comment]
I would say no user documentation change is required, but I'm not sure what the policy is about that.  The change affects compatibility with different browsers/browser versions.  Users aren't going to notice a change in behaviour, but web SAMP might work with this change when it doesn't in previous versions.

You can remove the URL from the change note if you want, I just thought that some traceability of the reason for the change would be useful.  But I don't understand the W3C process that has created this proposed protocol change well enough to know how long-lived that document is expected to be, so it's possible that this link could become out of date at some point.

**Review by tomdonaldson — APPROVED** [review]
Thanks @mbtaylor .  This looks good to me.

I committed the Change Log URL format change.   I think it's fine to have the URL there for now as it provides a quick jump to useful information in understanding this change.  If the URL dies, the link text is still specific enough to be searchable.

I looked briefly into adding a test for the change, but backed off when I saw:
- It wasn't immediately obvious to me how to adapt the existing test to check the preflight (OPTIONS) headers.
- I didn't look deeper because the relevant test is disabled in CI for flaky behavior (https://github.com/astropy/astropy/blob/62aaa295c33a1b653e0a35288f241db8bd89f3ba/astropy/samp/tests/test_web_profile.py#L28).
- I did manually verify in Chrome the desired behavior.

**Comment by pllim** [comment]
Thanks, all!

**Pull request merged**